### PR TITLE
SONAR-7478 WS api/users/identity_providers return external identity providers

### PR DESCRIPTION
**Pull request by teryk**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/user/ws/IdentityProvidersAction.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    IdentityProvidersWsResponse.Builder response = IdentityProvidersWsResponse.newBuilder();
+    response.addAllIdentityProviders(Lists.transform(identityProviderRepository.getAllEnabledAndSorted(), IdentityProviderToWsResponse.INSTANCE));
+    return response.build();
+  }
+
+  private enum IdentityProviderToWsResponse implements Function<IdentityProvider, WsUsers.IdentityProvider> {
+    INSTANCE;
+
+    @Override
+    public WsUsers.IdentityProvider apply(@Nonnull IdentityProvider input) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/user/ws/IdentityProvidersAction.java]: MAJOR Remove this "Nonnull" annotation to honor the overridden method's contract. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: server/sonar-server/src/main/java/org/sonar/server/user/ws/IdentityProvidersAction.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    this.identityProviderRepository = identityProviderRepository;
+  }
+
+  @Override
+  public void define(WebService.NewController context) {
+    context.createAction("identity_providers")
+      .setDescription("List the external identity providers")
+      .setResponseExample(getClass().getResource("identity_providers-example.json"))
+      .setSince("5.5")
+      .setInternal(true)
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/user/ws/IdentityProvidersAction.java]: it should not be internal

File: server/sonar-server/src/main/java/org/sonar/server/user/ws/IdentityProvidersAction.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  private enum IdentityProviderToWsResponse implements Function<IdentityProvider, WsUsers.IdentityProvider> {
+    INSTANCE;
+
+    @Override
+    public WsUsers.IdentityProvider apply(@Nonnull IdentityProvider input) {
+      WsUsers.IdentityProvider.Builder builder = WsUsers.IdentityProvider.newBuilder()
+        .setKey(input.getKey())
+        .setName(input.getName());
+      Display display = input.getDisplay();
+      if (display != null) {
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/user/ws/IdentityProvidersAction.java]: AFAIK, display can never be null

**Comment by julienlancelot** [comment]
Small feedbacks to apply, then LGTM

File: server/sonar-server/src/main/java/org/sonar/server/user/ws/IdentityProvidersAction.java (hunk 2 above)
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/user/ws/IdentityProvidersAction.java]: no need to make it public for the time being

File: server/sonar-server/src/main/java/org/sonar/server/user/ws/IdentityProvidersAction.java (hunk 3 above)
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/user/ws/IdentityProvidersAction.java]: ok

**Pull request merged**